core/src/main/java/com/yahoo/oak/NativeMemoryAllocator.java

```diff
@@ -52,13 +52,18 @@
     // A testable constructor
     NativeMemoryAllocator(long capacity, BlocksProvider blocksProvider) {
         this.blocksProvider = blocksProvider;
-        int blockArraySize = ((int) (capacity / blocksProvider.blockSize())) + 1;
-        // first entry of blocksArray is always empty
-        this.blocksArray = new Block[blockArraySize + 1];
+        this.capacity = capacity;
+        this.blocksArray = new Block[calcBlockArraySize()];
         // initially allocate one single block from pool
         // this may lazy initialize the pool and take time if this is the first call for the pool
         allocateNewCurrentBlock();
-        this.capacity = capacity;
+
+    }
+
+    private int calcBlockArraySize() {
```

Comment:
Where is it used?

Reply:
I noticed a bug: `clear()` does not re-allocate the block array. So I corrected the code. But since this allocation happens in `clear()` and in the constructor, I added this function to avoid code duplication.